website/models.py

```diff
@@ -648,22 +660,48 @@ class Project(models.Model):
     logo = models.ImageField(upload_to="project_logos", null=True, blank=True)
     created = models.DateTimeField(auto_now_add=True)
     modified = models.DateTimeField(auto_now=True)
+    contributors = models.ManyToManyField(
+        Contributor, related_name="projects", null=True, blank=True
+    )
 
     def __str__(self):
         return self.name
 
-    # # these need to be tested
-    # def get_github_logo_save_to_storage(self):
-    #     github_logo = requests.get(f"{self.github_url}/raw/master/logo.png")
-    #     if github_logo.status_code == 200:
-    #         self.logo.save(f"{self.slug}.png", ContentFile(github_logo.content))
-    #         self.save()
-    #         return self.logo.url
-
-    # def save(self, *args, **kwargs):
-    #     if not self.logo:
-    #         self.get_github_logo_save_to_storage()
-    #     super().save(*args, **kwargs)
-
-    def get_absolute_url(self):
-        return f"/project/{self.slug}"
+    def get_github_logo_save_to_storage(self, url):
+        owner = url.split("/")
+        link = "https://avatars.githubusercontent.com/" + owner[-2]
+        github_logo = requests.get(
+            link
+        )  # the image there is of the owner so we will have this as the logo
+        if github_logo.status_code == 200:
+            file_name = f"{self.slug}.png"
+            self.logo.save(file_name, ContentFile(github_logo.content), save=False)
+            return self.logo
+        return None
+
+    def get_contributors(self, github_url):
+        owner = github_url.split("/")
+        url = "https://api.github.com/repos/" + owner[-2] + "/" + owner[-1] + "/contributors"
```

Comment:
I suggest using https://github.com/PyGithub/PyGithub instead

Reply:
For this we require the access token so we need to apply for that and register this as a github app but this api i used here is used for both authorized and un-auth requests even by GitHub. Since our purpose is just to fetch the data and nothing complex i think we can go with this. Should i change this?

Reply:
No, not necessarily. You can do

`github_client = Github()` 

to use it as an anonymous